include/drivers/rtc.h

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file
+ * @brief RTC public API header file.
+ */
+/*
+ * Copyright (c) 2021 LACROIX Group
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <time.h>
+#include <device.h>
+// #include <dt-bindings/rtc/rtc.h>
+
+/**
+ * @brief RTC driver APIs
+ * @defgroup rtc_interface RTC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/** @brief Wake up timers enum. */
+enum rtc_wakeup_id {
+	RTC_WUT0,       /**< Wake Up timer 0. */
+	/* Add new item before RTC_WUT_NUM. */
+	RTC_WUT_NUM,    /**< Number of wut for the driver. */
+};
+
+/** @brief Alarms enum. */
+enum rtc_alarm_id {
+	RTC_ALARMA,     /**< Alarm A. */
+	RTC_ALARMB,     /**< Alarm B. */
+	/* Add new item before RTC_ALARM_NUM. */
+	RTC_ALARM_NUM,  /**< Number of alarms for the driver. */
+};
+
+/** @brief Specifies if the RTC Alarm is on date or Week day. */
+enum rtc_alarm_date_weekday {
+	RTC_ALARM_DATE_SEL,     /**< Alarm Date is selected, set parameter to a value in the 1-31 range. */
+	RTC_ALARM_WEEKDAY_SEL,  /**< Alarm Weekday is selected, set parameter to a value in the 0-6 range (days since Sunday). */
+};
+
+/** @brief Specifies the RTC Alarm Masks. */
+enum rtc_alarm_mask {
+	RTC_ALARM_MASK_NONE,            /**< No mask. */
+	RTC_ALARM_MASK_DATE_WEEKDAY,    /**< Mask on date or weekday. */
+	RTC_ALARM_MASK_HOURS,           /**< Mask on hours. */
+	RTC_ALARM_MASK_MIN,             /**< Mask on minutes. */
+	RTC_ALARM_MASK_SEC,             /**< Mask on seconds. */
+	RTC_ALARM_MASK_ALL,             /**< Mask on all. */
+};
+
+/** @brief Wake up timer structure. */
+struct rtc_wakeup {
+	uint32_t period;                                /**< Wake up timer period in milliseconds. */
+	void (*callback)(const struct device *dev);     /**< Callback function. Can be NULL. */
```

Comment:
We should provide a user_data variable with the alarm callback to allow for multiple instances of drivers, or C++ classes, which will share the same static callback for different alarms. They can provide pointers to themselves in the user_data field.

```
void (*callback)(const struct device *dev, void *user_data);   /**< Callback function. Can be NULL. */
```

We should also add a boolean to choose between periodic wakeup timer and singleshot wakeup timer
```
    bool periodic; /**< Set if wakeup timer should be periodic*/
```
Or the inverse if the default is periodic
```
    bool singleshot; /**< Set if wakeup timer should be single shot*/
```